Makefile

```diff
@@ -406,7 +406,11 @@ $(GINKGO): $(LOCALBIN)
 
 .PHONY: test-e2e
 test-e2e: ginkgo ## Run end-to-end tests. See the 'tests/e2e/README.md' file for more details.
-	$(GINKGO) $(GINKGO_FLAGS) tests/e2e
+	$(GINKGO) $(GINKGO_FLAGS) --label-filter="!upgrade" tests/e2e
```

Comment:
Curious: What is the reason to separate it this way?

Reply:
The "normal" tests have a `BeforeSuite` logic that deploys the operator image once in the cluster (from an end-user perspective) before running the tests.
The "upgrade" test is different because it needs a different version (older) version of the operator to be installed. As such, it cannot share the same controller as the other ones.

Reply:
OK, then probably it makes sense to implement it in separate file?
To me it makes sense to separate it b/c I suppose there are no reason to call it unless you change default configuration (more rare case than regression caused by code change), so it can be release related at all.

Reply:
Well, the original issue was caused by a regression due to some code change (change to the DB Service object spec).
But we should capture those upgrade issues early on, be it release or code-related, and that's why I think it will be helpful to have them run regularly.
I am using a label decorator here as a way to separate those, but okay, I can move those to a dedicated spec file.

Reply:
Moved to a separate spec file. See [`077da8a` (#384)](https://github.com/janus-idp/operator/pull/384/commits/077da8a60c37adde9a978562b7e7559932cbce02)